src/System.Management.Automation/engine/Modules/GetModuleCommand.cs

```diff
@@ -465,14 +459,10 @@ private void GetAvailableViaPsrpSession(string[] names, IDictionary<string, Modu
 
         private void GetAvailableLocallyModules(string[] names, IDictionary<string, ModuleSpecification> moduleSpecTable, bool all)
         {
-            var refresh = Refresh.IsPresent;
-            var modules = GetModule(names, all, refresh);
-
-            foreach (
-                var psModule in
-                    modules.Where(module => ModuleMatch(module, moduleSpecTable, PSEdition)).Select(module => new PSObject(module))
-                )
+            IEnumerable<PSModuleInfo> modules = GetModule(names, all, Refresh);
```

Comment:
Why we use `IEnumerable`? We could use List<T> - seems it is more thriftily.

Reply:
Used `IEnumerable` here because:
* The interface type defines all the functionality we need without added constraints
* We don't require indexing or even size checking, just ordered enumeration
* The method is free to implement the collection as it wishes

*But*, perhaps there are benefits to using a `List` or at least an `IList`?
* Less type complexity (and binding to a concrete type not an issue with an internal API)?
* Provides strictness and ordering guarantees